gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/artifacts/ExtractSingleFileOrManifest.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.dist.artifacts;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.util.jar.Manifest;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+import org.gradle.api.artifacts.transform.CacheableTransform;
+import org.gradle.api.artifacts.transform.InputArtifact;
+import org.gradle.api.artifacts.transform.TransformAction;
+import org.gradle.api.artifacts.transform.TransformOutputs;
+import org.gradle.api.file.FileSystemLocation;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.tasks.PathSensitive;
+import org.gradle.api.tasks.PathSensitivity;
+import org.gradle.util.GFileUtils;
+
+@CacheableTransform
+public abstract class ExtractSingleFileOrManifest implements TransformAction<FileAndManifestExtractParameter> {
+    private static final String MANIFEST = "META-INF/MANIFEST.MF";
+
+    @PathSensitive(PathSensitivity.NAME_ONLY)
+    @InputArtifact
+    public abstract Provider<FileSystemLocation> getInputArtifact();
+
+    @Override
+    public final void transform(TransformOutputs outputs) {
+        File jarFile = getInputArtifact().get().getAsFile();
+        String pathToExtract = getParameters().getPathToExtract().get();
+        String manifestKey = getParameters().getKeyToExtract().get();
+
+        try (ZipFile zipFile = new ZipFile(jarFile)) {
```

Comment:
We're probably going to need to update dev-env to support get pdeps from resources as well as manifest too: https://github.palantir.build/foundry/dev-env/blob/e690c9a01b68297b59269febca4f4a9df0ff8dcc/dev-env-common/src/main/java/com/palantir/devenv/sls/PdepJarExtractor.java.

I wonder if we should make that library to reduce duplication.

Reply:
the existing `gradle-recommended-product-dependencies` should expose enough to make it easy for people to consistently access pdeps however they are embedded